### PR TITLE
QChem: fixes for 5.4 unit tests

### LINKED ISSUE #977: Update Q-Chem unit tests to 5.4

### PR DESCRIPTION
**Pull request by berquist**
A continuation of #1072. Closes #977.

### CODE REVIEW

File: cclib/parser/qchemparser.py [hunk 1]
```diff
@@ -772,6 +778,21 @@ def extract(self, inputfile, line):
                     self.norbdisp_beta = min(self.norbdisp_beta, self.nbasis)
                     self.norbdisp_beta_aonames = min(self.norbdisp_beta_aonames, self.nbasis)
 
+            # Finally, versions of Q-Chem greater than 5.1 print all MOs in
```
Comment (berquist) [inline cclib/parser/qchemparser.py:781]: I have access to more fine-grained versions of Q-Chem, so I'm going to try refining this. But really the whole block seems undesirable...

**Comment by berquist** [comment]
Winter cleaning gone wrong...

**Comment by berquist** [comment]
This and the associated data update (https://github.com/cclib/cclib-data/pull/157) are ready for review.

File: test/regression.py [hunk 2]
```diff
@@ -3353,12 +3366,33 @@ def testatommasses(self):
...
     "Psi4/Psi4-beta5/water_mp2.out":    GenericMP2Test,
 
+    "QChem/QChem4.2/C_bigbasis.out":         QChemBigBasisTest,
+    "QChem/QChem4.2/MoOCl4_sp.out":          GenericCoreTest,
+    "QChem/QChem4.2/Trp_polar.out":          ReferencePolarTest,
+    "QChem/QChem4.2/dvb_bomd.out":           GenericBOMDTest,
+    "QChem/QChem4.2/dvb_gopt.out":           GenericGeoOptTest,
+    "QChem/QChem4.2/dvb_ir.out":             QChemIRTest,
+    "QChem/QChem4.2/dvb_raman.out":          QChemRamanTest,
+    # also GenericBasisTest
```
Comment (shivupa) [inline test/regression.py:3376]: Can you explain what these comments are referring to?

File: test/regression.py [hunk 3]
```diff
@@ -3353,12 +3366,33 @@ def testatommasses(self):
...
+    "QChem/QChem4.2/dvb_sp.out":             GenericSPTest,
+    "QChem/QChem4.2/dvb_sp_un.out":          GenericSPunTest,
+    "QChem/QChem4.2/dvb_td.out":             QChemTDDFTTest,
+    "QChem/QChem4.2/water_ccd.out":          GenericCCTest,
+    "QChem/QChem4.2/water_ccsd(t).out":      GenericCCTest,
+    "QChem/QChem4.2/water_ccsd.out":         GenericCCTest,
+    "QChem/QChem4.2/water_cis.out":          QChemCISTest,
+    # ???
+    # create mode 100644 QChem/QChem4.2/water_ir.out
+    # create mode 100644 QChem/QChem4.2/water_ir_anharm.out
```
Comment (shivupa) [inline test/regression.py:3386]: Here too.

Comment (berquist) [reply test/regression.py:3386]: These were files copied over that didn't correspond to any tests in `testdata`. I think they may be from when we were thinking about parsing anharmonic frequencies. I've just left them in in the regressions repo.

File: test/regression.py (hunk 2 above)
Comment (berquist) [reply test/regression.py]: `dvb_sp.out` is always used for at least the `Basis` and `SP` tests.  If you consider this being a dict from filename to unit test class, then you can only have one test per file.  If you look there are actually a lot of duplicated keys which will lead to skipped unit tests.

**Pull request merged**